File: part1.m
%% Part 1: Matrix Operations for a Feedforward Network
clear all

% M is the number of rows, and N is the number of columns.
M = 10;
N = 50;

% Generate a random number between 0 and 1, drawed from a uniform 
% distribution.
W = rand( M, N );

% Map the numbers between [0,1] to numbers between [-1,1].
W = 2 .* ( W - 0.5 );

% Generate a column input vector x with 50 elements.
x = randn( 50, 1 );

% Compute the output of the network.
y = W * x;

File: part1.py
#!/usr/bin/env python

import numpy as np # this package is used for most mathematical operations
import matplotlib.pyplot as plt

# define input/output dimensions
M = 10
N = 50

# define weight matrix
W = np.random.rand(M,N)
W = 2*(W-0.5)

# define input vector
x = np.random.rand(N) # random is the numpy package used to generate random numbers
y = W*x
